src/components/tx-flow/flows/TokenTransfer/ReviewSpendingLimitTx.tsx

```diff
@@ -96,9 +99,14 @@ const ReviewSpendingLimitTx = ({
       setTxFlow(undefined)
     } catch (_err) {
       const err = asError(_err)
-      logError(Errors._801, err)
-      setIsSubmittable(true)
-      setSubmitError(err)
+      if ('code' in err && err.code === ErrorCode.ACTION_REJECTED) {
+        setIsSubmittable(true)
+        setIsRejectedByUser(true)
+      } else {
+        logError(Errors._801, err)
+        setIsSubmittable(true)
+        setSubmitError(err)
+      }
```

Comment:
We can reuse `isUserRejection` here and pull out `setIsSubmittable` from the condition since its set to true in both cases. This can also be done for the same block in `SignForm` and `ExecuteForm`.